internal/bminventory/inventory.go

```diff
@@ -1740,14 +1740,14 @@ func (b *bareMetalInventory) generateClusterInstallConfig(ctx context.Context, c
 		return errors.Wrapf(err, "failed to get install config for cluster %s", cluster.ID)
 	}
 
-	releaseImage, err := b.versionsHandler.GetReleaseImage(cluster.OpenshiftVersion)
-
+	ocpVersion, err := b.versionsHandler.GetOpenshiftVersion(cluster.OpenshiftVersion)
 	if err != nil {
-		log.WithError(err).Errorf("failed to get release image for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
-		return errors.Wrapf(err, "failed to get release image for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
+		msg := fmt.Sprintf("failed to get OpenshiftVersion for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
+		log.WithError(err).Errorf(msg)
+		return errors.Wrapf(err, msg)
```

Comment:
why log the error and then return it upwards?
seems like something that only leads to printing the same error on two places
iiuc just returning it with the wrapping is enough

Reply:
We actually do the same for the other errors in generateClusterInstallConfig, need to check if redundant.

Reply:
Verified that the log is indeed printed only once. So the log seems mandatory.